drivers/SerialBase.h

```diff
@@ -291,6 +291,30 @@ class SerialBase : private NonCopyable<SerialBase> {
     int _base_getc();
     int _base_putc(int c);
 
+    /** Initialize serial port
+     */
+    void _init();
+
+    /** Deinitialize serial port
+     */
+    void _deinit();
+
+    /** Enable serial input
+     *
+     * If both serial input and serial output are disabled, the
+     * peripheral is freed. If either serial input or serial
+     * output is re-enabled, the peripheral is reinitialized.
+     */
+    void _enable_input(bool enable = true);
```

Comment:
This should be just `enable_input` and should incorporate the `lock` calls. It can be part of the proper public API for `SerialBase` and hence `Serial`, just like it currently is for `UARTSerial`.  Part of the point of this is to provide that API.

For `Serial` and `RawSerial` it will be inherited, as `SerialBase` is the public base for them. It will be calling the virtual `lock`, which will then be the mutex lock for `Serial` and nothing for `RawSerial`.

Basically like `set_break` or `write`.

Reply:
Note that `Serial` will need a `using SerialBase::enable_input;` in its class definition to make sure it picks up that implementation - it will be inheriting both that one and the dummy one from `FileHandle`, which will produce an ambiguity compile error.